src/chunkserver/network_worker_thread.cc

```diff
@@ -285,125 +288,128 @@ void worker_delayed_close(uint8_t status, void *e) {
-		eptr->messageSerializer->serializeCstoclReadStatus(
-		    buffer, eptr->chunkId, SAUNAFS_STATUS_OK);
-		worker_create_attached_packet(eptr, buffer);
-		sassert(eptr->isChunkOpen);
-		job_close(eptr->workerJobPool, nullptr, nullptr, eptr->chunkId,
-		          eptr->chunkType);
-		eptr->isChunkOpen = 0;
+		messageSerializer->serializeCstoclReadStatus(
+		    buffer, chunkId, SAUNAFS_STATUS_OK);
+		createAttachedPacket(buffer);
+		sassert(isChunkOpen);
+
+		job_close(workerJobPool, nullptr, nullptr, chunkId,
+		          chunkType);
+		isChunkOpen = 0;
 		// no error - do not disconnect - go direct to the IDLE state, ready for
 		// requests on the same connection
-		eptr->state = ChunkserverEntry::State::Idle;
-		LOG_AVG_STOP(eptr->readOperationTimer);
+		state = State::Idle;
+		LOG_AVG_STOP(readOperationTimer);
 	} else {
-		const uint32_t totalRequestSize = eptr->size;
-		const uint32_t thisPartOffset = eptr->offset % SFSBLOCKSIZE;
+		const uint32_t totalRequestSize = size;
+		const uint32_t thisPartOffset = offset % SFSBLOCKSIZE;
 		const uint32_t thisPartSize = std::min<uint32_t>(
 				totalRequestSize, SFSBLOCKSIZE - thisPartOffset);
 		const uint16_t totalRequestBlocks =
-				(totalRequestSize + thisPartOffset + SFSBLOCKSIZE - 1) / SFSBLOCKSIZE;
+		    (totalRequestSize + thisPartOffset + SFSBLOCKSIZE - 1) /
+		    SFSBLOCKSIZE;
+
 		std::vector<uint8_t> readDataPrefix;
-		eptr->messageSerializer->serializePrefixOfCstoclReadData(
-		    readDataPrefix, eptr->chunkId, eptr->offset, thisPartSize);
-		auto packet =
-		    worker_create_detached_packet_with_output_buffer(readDataPrefix);
+		messageSerializer->serializePrefixOfCstoclReadData(
+		    readDataPrefix, chunkId, offset, thisPartSize);
+		auto packet = createDetachedPacketWithOutputBuffer(readDataPrefix);
 		if (packet == nullptr) {
-			eptr->state = ChunkserverEntry::State::Close;
+			state = State::Close;
 			return;
 		}
-		eptr->readPacket = std::move(packet);
+		readPacket = std::move(packet);
+
 		uint32_t readAheadBlocks = 0;
 		uint32_t maxReadBehindBlocks = 0;
-		if (!eptr->isChunkOpen) {
+
+		if (!static_cast<bool>(isChunkOpen)) {
 			if (gHDDReadAhead.blocksToBeReadAhead() > 0) {
-				readAheadBlocks = totalRequestBlocks + gHDDReadAhead.blocksToBeReadAhead();
+				readAheadBlocks =
+				    totalRequestBlocks + gHDDReadAhead.blocksToBeReadAhead();
 			}
 			// Try not to influence slow streams to much:
 			maxReadBehindBlocks = std::min(totalRequestBlocks,
 					gHDDReadAhead.maxBlocksToBeReadBehind());
 		}
-		eptr->readJobId = job_read(
-		    eptr->workerJobPool, worker_read_finished, eptr, eptr->chunkId,
-		    eptr->chunkVersion, eptr->chunkType, eptr->offset, thisPartSize,
-		    maxReadBehindBlocks, readAheadBlocks,
-		    eptr->readPacket->outputBuffer.get(), !eptr->isChunkOpen);
-		if (eptr->readJobId == 0) {
-			eptr->state = ChunkserverEntry::State::Close;
+
+		readJobId = job_read(workerJobPool, readFinishedCallback, this, chunkId,
```

Comment:
Consider use lambda expression to avoid passing this as argument.
eg:
```
writeJobId = job_write(workerJobPool, 
    [this](uint8_t status, void *entry) { this->writeFinishedCallback(status, entry); },
    nullptr, opChunkId, chunkVersion, chunkType, blocknum, opOffset, opSize, crc, dataToWrite);
```